### PR TITLE
PL - Dependencies: Remediate Babel security issue

### PR DESCRIPTION
**Pull request by mejiaj**
## Description

Removing deprecated dependencies and redundant JavaScript. Closes https://github.com/GSA/plainlanguage.gov/security/dependabot/49.

### Remove deprecated dependencies

Removed the following deprecated dependencies:
- [babel-preset-es2015 - npm](https://www.npmjs.com/package/babel-preset-es2015)
- [babel-plugin-transform-es2015-modules-commonjs - npm](https://www.npmjs.com/package/babel-plugin-transform-es2015-modules-commonjs)

### Create NVMRC

Added an NVMRC file to always use the latest LTS version of node. This will help us stay on top of node versions.

### Remove sticky polyfill

The only custom JS included was setting an explicit height to prevent a slight jump on hover. Position sticky has good support and didn't notice any issues on pages with sticky sidenav, like [`law/agency-programs`](https://federalist-dad55c1b-07c6-4c7e-89da-5aff1f9335fd.sites.pages.cloud.gov/preview/gsa/plainlanguage.gov/jm-poam-babel-update/law/agency-programs/).

## Testing

- [ ] Running `npm run build-js` builds without errors.
- [ ] Running `npm audit` shows zero vulnerabilities.
- [ ] Sticky navigation in works and there isn't a jump on hover | [preview link →](https://federalist-dad55c1b-07c6-4c7e-89da-5aff1f9335fd.sites.pages.cloud.gov/preview/gsa/plainlanguage.gov/jm-poam-babel-update/law/agency-programs/).

## Impact

### CODE REVIEW

File: assets/js/main.js
Comment (mejiaj) [inline assets/js/main.js:1]: > [!TIP]
> File is left empty to allow custom JS in the future. Also avoids a build error. The polyfill and jQuery are no longer needed.

File: package.json [hunk 1]
```diff
@@ -24,17 +24,13 @@
   },
   "homepage": "https://github.com/GSA/plainlanguage.gov#readme",
   "dependencies": {
-    "@babel/core": "^7.24.4",
-    "@babel/preset-env": "^7.24.4",
+    "@babel/core": "^7.24.9",
+    "@babel/preset-env": "^7.24.8",
     "babel-loader": "^8.2.3",
-    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.2",
```
Comment (mejiaj) [inline package.json:30]: > [!TIP]
> Now included in `@babel/preset-env`. 
[Source →](https://babeljs.io/docs/babel-plugin-transform-modules-commonjs)

File: package.json [hunk 2]
```diff
@@ -24,17 +24,13 @@
...
-    "babel-preset-es2015": "^6.14.0",
```
Comment (mejiaj) [inline package.json:31]: > [!TIP]
> Replaced by `@babel/preset-env`.
[Source →](https://babeljs.io/docs/babel-preset-es2015)

File: package.json [hunk 3]
```diff
@@ -24,17 +24,13 @@
...
     "basscss-sass": "^4.0.0",
     "glob-parent": "^6.0.2",
-    "jquery": "^3.7.1",
     "pa11y-ci": "^2.4.2",
-    "stickyfilljs": "^2.1.0",
```
Comment (mejiaj) [inline package.json:36]: > [!TIP]
> jQuery and stickyfill no longer needed as position sticky is well supported.

**Review by mejiaj — COMMENTED** [review]
Added comments for clarity.

**Review by clmedders — APPROVED** [review]

**Pull request merged**